mesh/mesh.hpp

```diff
@@ -91,6 +91,64 @@ class Mesh
    Array<Element *> boundary;
    Array<Element *> faces;
 
+   /** @brief This structure stores the low level information necessary to
+       interpret the configuration of elements on a specific face. This
+       information is accessed through Mesh::GetFaceElements and
+       Mesh::GetFaceInfos.
+       For accessing higher level deciphered information
+       look at Mesh::FaceInformation, and its accessor Mesh::GetFaceInformation.
+
+       Each face contains information on the indices, local reference faces,
+       orientations, and potential non-conformity for the two neighboring
+       elements on a face.
+       Each face can either be an interior, boundary, or shared interior face.
+       Each interior face is shared by two elements referred as Elem1 and Elem2.
+       For boundary faces only the information on Elem1 is relevant.
+       Shared interior faces correspond to faces where Elem1 and Elem2 are
+       distributed on different MPI ranks.
+       Regarding conformity, three cases are distinguished, conforming faces,
+       non-conforming slave faces, and non-conforming master faces. Master and
+       slave referring to the coarse and fine elements respectively on a
+       non-conforming face.
+       Non-conforming slave faces always have the slave element as Elem1 and
+       the master element as Elem2. On the other side, non-conforming master
+       faces always have the master element as Elem1, and one of the slave
+       element as Elem2. Except for ghost non-conforming slave faces, where
+       Elem1 is the master side and Elem2 is the slave side.
+
+       The indices of Elem1 and Elem2 can be indirectly extracted from
+       FaceInfo::Elem1No and FaceInfo::Elem2No, read the note below for special
+       cases on the index of Elem2.
+
+       The local face identifiants are deciphered from FaceInfo::Elem1Inf and
```

Comment:
Nice documentation, but identifiant is not an English word. Do you mean index or identity?

Reply:
ID